android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageMigration.java

```diff
@@ -0,0 +1,161 @@
+package com.reactnativecommunity.asyncstorage;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.RequiresApi;
+
+import com.facebook.react.modules.storage.ReactDatabaseSupplier;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+
+// A utility class that migrates a scoped AsyncStorage database to RKStorage.
+// This utility only runs if the RKStorage file has not been created yet.
+public class AsyncStorageMigration {
+    static final String LOG_TAG = "ScopedStorageMigration";
+
+    private static Context mContext;
+
+    public static void migrate(Context context) {
+        mContext = context;
+
+        // Only migrate if the default async storage file does not exist.
+        if (isAsyncStorageDatabaseCreated()) {
+            return;
+        }
+
+        ArrayList<File> expoDatabases = getExpoDatabases();
+
+        File expoDatabase = getLastModifiedFile(expoDatabases);
+
+        if (expoDatabase == null) {
+            Log.v(LOG_TAG, "No scoped database found");
+            return;
+        }
+
+        try {
+            // Create the storage file
+            ReactDatabaseSupplier.getInstance(mContext).get();
+            copyFile(new FileInputStream(expoDatabase), new FileOutputStream(mContext.getDatabasePath(ReactDatabaseSupplier.DATABASE_NAME)));
+            Log.v(LOG_TAG, "Migrated most recently modified database " + expoDatabase.getName() + " to RKStorage");
+        } catch (Exception e) {
+            Log.v(LOG_TAG, "Failed to migrate scoped database " + expoDatabase.getName());
+            e.printStackTrace();
+            return;
+        }
+
+        try {
+            for (File file : expoDatabases) {
+                if (file.delete()) {
+                    Log.v(LOG_TAG, "Deleted scoped database " + file.getName());
```

Comment:
let's leave it in place rather than deleting

Reply:
changed my mind and it should delete, to behave the same as the ios script. thanks for reverting :)